test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -689,17 +591,150 @@ private static void CounterUpdateThread<T>(object obj)
 
             for (int i = 0; i < numberOfMetricUpdateByEachThread; i++)
             {
-                counter.Add(valueToUpdate, new KeyValuePair<string, object>("verb", "GET"));
+                counter.Add(arguments.DeltaValueUpdatedByEachCall);
+            }
+        }
+
+        private static void HistogramUpdateThread<T>(object obj)
+            where T : struct, IComparable
+        {
+            var arguments = obj as UpdateThreadArguments<T>;
+            if (arguments == null)
+            {
+                throw new Exception("Invalid args");
             }
+
+            var mre = arguments.MreToBlockUpdateThread;
+            var mreToEnsureAllThreadsStart = arguments.MreToEnsureAllThreadsStart;
+            var histogram = arguments.Instrument as Histogram<T>;
+
+            if (Interlocked.Increment(ref arguments.ThreadsStartedCount) == numberOfThreads)
+            {
+                mreToEnsureAllThreadsStart.Set();
+            }
+
+            // Wait until signalled to start calling update on aggregator
+            mre.WaitOne();
+
+            for (int i = 0; i < numberOfMetricUpdateByEachThread; i++)
+            {
+                histogram.Record(arguments.DeltaValueUpdatedByEachCall);
```

Comment:
to make it more interesting, we should do different values, and a well distributed bucket.

Reply:
Made each thread record data to each of the default buckets